rtic-macros/src/codegen/pre_init.rs

```diff
@@ -11,10 +11,16 @@ pub fn codegen(app: &App, analysis: &Analysis) -> Vec<TokenStream2> {
     // Disable interrupts -- `init` must run with interrupts disabled
     stmts.push(quote!(rtic::export::interrupt::disable();));
 
+    // TODO dirty patch, needs to be addressed
+    #[cfg(not(feature = "riscv-slic"))]
```

Comment:
Maybe the following line could come from a new binding?

Reply:
That's a good idea. We can use bindings to partially remove the Peripherals struct depending on the backend.